Modules/Options/Options.lua

```diff
@@ -47,6 +47,10 @@ local FEED_MINIMAL = "FEED_MINIMAL"
 local FEED_NORMAL = "FEED_NORMAL"
 local FEED_VERBOSE = "FEED_VERBOSE"
 
+-- Tooltip activation
+local TIP_ACTIVATION_HOVER = "TIP_ACTIVATION_HOVER"
+local TIP_ACTIVATION_CLICK = "TIP_ACTIVATION_CLICK"
+
```

Comment:
Shouldn't this be read from the SharedConstants DB? This way it's defined in just one place and not three

Reply:
That might be a good idea (if sharing the contstants from the main module is straightforward), but then that should probably be a separate refactor since it looks to me like a bunch of other things are also double defined in the options module.

Reply:
I figured it's a trivial change so we might as well do it right the first time around. Just because the original author copy/pasted constants everywhere, doesn't mean we have to keep doing it :)

They're exported via ``Rarity.CONSTANTS`` in ``Core.lua``. You can simply access them globally in the Options module. I can do this myself, if you like, but it's probably easier to add it to the PR for you and it'll be more cohesive, too.

Reply:
No problem, I'll push these changes tomorrow.

Reply:
Ok, I'm having some issues with using `Rarity.CONSTANTS` in the options module. It just throws `table index is nil` when I try to access one of the values from there.

So if instead of

```lua
local TIP_ACTIVATION_HOVER = "TIP_ACTIVATION_HOVER"
local TIP_ACTIVATION_CLICK = "TIP_ACTIVATION_CLICK"
```

I did (not final, just example)

```lua
local ACTIVATION_METHOD_HOVER = Rarity.CONSTANTS.ACTIVATION_METHOD_HOVER
local ACTIVATION_METHOD_CLICK = Rarity.CONSTANTS.ACTIVATION_METHOD_CLICK
```

it would throw `table index is nil` where it's used

```lua
tooltipActivation = {
  type = "select",
  name = L["Tooltip activation"],
  desc = L["If \"On click\" is selected, activating the tracker is done via CTRL + SHIFT + Click, otherwise it's activated with a simple click."],
  values = { -- table index is nil
	  [ACTIVATION_METHOD_HOVER] = L["On hover"], -- table index is nil
	  [ACTIVATION_METHOD_CLICK] = L["On click"], -- table index is nil
  }, -- table index is nil
  get = function() return self.db.profile.tooltipActivation end,
  set = function(info, val)
	  self.db.profile.tooltipActivation = val
	  self:Update("OPTIONS")
  end,
  order = newOrder(),
}, -- tooltipActivation
```

Any thoughts? Is it an execution order thing or something else? `Rarity.CONSTANTS` seem to be defined at least, as far as I can figure.

Debugging is turning out to be a bit of an issue since I'm both new to LUA and addon dev. :P

Reply:
The options module is (optionally) loaded on-demand, by the Rarity addon itself, so execution order isn't a problem here. However, the constants for your setting reside in the ``Rarity.CONSTANTS.TOOLTIP`` table, as defined in ``SharedConstants.lua`` (``C`` is assigned ``Rarity.CONSTANTS`` for convenience).

You can debug table using ``/tinspect <tableName>`` or just search manually. This utility is provided by Blizzard and works better than doing ``/dump <tableName>`` manually, which is another option.

Reply:
Don't you just love it when you're an idiot? :D